components/lib/push-firebase/src/main/java/mozilla/components/lib/push/firebase/AbstractFirebasePushService.kt

```diff
@@ -59,4 +70,17 @@ abstract class AbstractFirebasePushService : FirebaseMessagingService(), PushSer
         FirebaseMessaging.getInstance().isAutoInitEnabled = false
         stopSelf()
     }
+
+    /**
+     * Removes the Firebase instance ID, thus forcing a new messaging service token to be generated.
+     */
+    override fun forceRegistrationRenewal() {
```

Comment:
However, we'd still need to have this method for when `verifyConnection` call says we're not connected properly, and need to get a new token. So this is probably fine.

Reply:
We need something at the service level since each service is different, but I see your point though. Maybe on the service we can have an "unregister" and on the feature we call: stop, unregister, start (or wtv it requires)?

> However, we'd still need to have this method for when verifyConnection call says we're not connected properly, and need to get a new token.

This part isn't related though. When verifyConnection says we're not connected properly, it only checks if our subscriptions have issues, so we _re-subscribe_ for that but we don't _re-register_.

Reply:
> This part isn't related though. When verifyConnection says we're not connected properly, it only checks if our subscriptions have issues, so we re-subscribe for that but we don't re-register.

I thought it was also used to indicate if the client can't be reached via the existing registration information? Which could often mean "bad bridged push service token, give me a new one". Or am I wrong?

Reply:
It's only for subscription issues from what I re-call: 

Docs: https://github.com/mozilla/application-services/blob/master/components/push/src/communications.rs#L68-L69
Impl: https://github.com/mozilla/application-services/blob/master/components/push/src/subscriber.rs#L124-L138

The idea was to use `verifyConnection` to notify Gecko for web content that listens for `onPushSubscriptionChange`.

I can bring this up again with a-s but maybe the bad bridge failure is what [this TODO refers to](https://github.com/mozilla/application-services/blob/master/components/push/src/communications.rs#L54) since we had pushed it back to a future version.

Reply:
Re: the bad bridged push service, I've filed this issue as follow up work to connect the FxA device constellation to the push feature.